eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -132,6 +132,31 @@ jobs:
   - checkout: self
     submodules: recursive
 
+  # Validate all the files are formatted correctly according to the .clang-format file.
+  - bash: |
+      # Run clang-format recursively on each source and header file within the repo sdk folder.
+      echo "Check clang-formatting (will use version 10)"
+      sudo apt-get update > /dev/null
+      sudo apt-get install clang-format-10 > /dev/null
+      clang-format-10 --version
+      find sdk/ -regex '.*\.\(cpp\|hpp\)' -exec clang-format-10 -style=file -i {} \;
+
+      if [[ `git status | grep modified | awk '{print $2}'` ]]; then
+        echo Some files were not formatted correctly according to the .clang-format file.
+        echo Please run clang-format version 10 or greater to fix the issue by using this bash command at the root of the repo:
+        echo "find sdk/ -regex '.*\.\(cpp\|hpp\)' -exec clang-format -style=file -i {} \;"
```

Comment:
Just to confirm, if I run this command locally right after this PR is merged, will there be any code formatting difference?
`find sdk/ -regex '.*\.\(cpp\|hpp\)' -exec clang-format -i {} \;`

How can I run a similar command on Windows (VS dev prompt for instance), or does it have to bash? I ran it in bash, and that works.

Not sure if adding the `/sdk` directory is necessary.

Reply:
I think the powerShell supports `find` command.  

However, if you are on VS, you should rely on VS using the .clang-format file to do formatting. 

The /sdk is to speed up a little bit in the case you have already generated the project.  For instance, if you generate debug and release projects on Windows, you will get the folder out/ with all the source files to be built.  Then you will be checking its format.
That's why we only care about ./sdk content

Reply:
> The /sdk is to speed up a little bit in the case you have already generated the project.

Nice, great idea!

Reply:
> However, if you are on VS, you should rely on VS using the .clang-format file to do formatting.

I don't know how to use VS to format all the files in one shot. I tend to run Ctrl+K, Ctrl+D on the files I modify to do the formatting in VS. Got any suggestions on how I can format all files?

Reply:
> Got any suggestions on how I can format all files?

Well, when working with VS, why would you need to run formatting to all files?
I think you would need to rely on some 3rd party ext for it, at any case. LIke https://marketplace.visualstudio.com/items?itemName=munyabe.FormatAllFiles